dora/core/client/fs/src/main/java/alluxio/client/file/dora/WorkerLocationPolicy.java

```diff
@@ -57,7 +61,7 @@ public WorkerLocationPolicy(int numVirtualNodes) {
   public List<BlockWorkerInfo> getPreferredWorkers(List<BlockWorkerInfo> blockWorkerInfos,
```

Comment:
[May not need to fix in this PR]
I find the semantic very confusing when you can return an empty list here. Does it mean "there are no workers" or "there are workers but I simply don't prefer any ones"?

If there's only one method here, I prefer it throws an error when there are no workers. There's `ExceptionMessage.NO_WORKER_AVAILABLE` for that use.

If someone (like the job scheduler) really wants to tolerate that situation, I guess we just create two separate methods:
```
public List<> mustFindPreferredWorkers() { // throw when no workers available }

public List<> maybeFindPreferredWorkers() { // do not throw when no workers available }
```

Reply:
@jja725 I see your CopyJob/MoveJob are using this method. Do you think the change in semantics can work?

Reply:
I think we should separate the logic of refreshing the worker list and that of getting a preferred worker. The responsibility of refreshing the worker list should go to the membership service.